cmake/externals/triton.cmake

```diff
@@ -44,36 +36,44 @@ if (WIN32)
   add_custom_target(vcpkg_integrate ALL DEPENDS vcpkg_integrate.stamp)
   set(VCPKG_DEPENDENCIES "vcpkg_integrate")
 
-  # use static-md so it adjusts for debug/release CRT
-  # https://stackoverflow.com/questions/67258905/vcpkg-difference-between-windows-windows-static-and-other
   function(vcpkg_install PACKAGE_NAME)
     add_custom_command(
-      OUTPUT ${VCPKG_SRC}/packages/${PACKAGE_NAME}_${vcpkg_target_platform}-windows-static-md/BUILD_INFO
+      OUTPUT ${VCPKG_SRC}/packages/${PACKAGE_NAME}_${vcpkg_triplet}/BUILD_INFO
       COMMAND ${CMAKE_COMMAND} -E echo ${VCPKG_SRC}/vcpkg install --vcpkg-root=$ENV{VCPKG_ROOT}
-                                  ${PACKAGE_NAME}:${vcpkg_target_platform}-windows-static-md
+                                  ${PACKAGE_NAME}:${vcpkg_triplet}
       COMMAND ${VCPKG_SRC}/vcpkg install --vcpkg-root=$ENV{VCPKG_ROOT}
-                                         ${PACKAGE_NAME}:${vcpkg_target_platform}-windows-static-md
+                                         ${PACKAGE_NAME}:${vcpkg_triplet}
       WORKING_DIRECTORY ${VCPKG_SRC}
       DEPENDS vcpkg_integrate)
 
     add_custom_target(
       get${PACKAGE_NAME}
       ALL
-      DEPENDS ${VCPKG_SRC}/packages/${PACKAGE_NAME}_${vcpkg_target_platform}-windows-static-md/BUILD_INFO)
+      DEPENDS ${VCPKG_SRC}/packages/${PACKAGE_NAME}_${vcpkg_triplet}/BUILD_INFO)
 
     list(APPEND VCPKG_DEPENDENCIES "get${PACKAGE_NAME}")
     set(VCPKG_DEPENDENCIES ${VCPKG_DEPENDENCIES} PARENT_SCOPE)
   endfunction()
 
-  vcpkg_install(rapidjson)
+  vcpkg_install(rapidjson)  # required by triton
   vcpkg_install(openssl)
   vcpkg_install(curl)
 
-  set(triton_extra_cmake_args -DVCPKG_TARGET_TRIPLET=${vcpkg_target_platform}-windows-static-md
+  # fake dependency between openssl and rapidjson.
+  # without this 2 `vcpkg install` commands run in parallel which results in errors like this on the first build:
+  #  write_contents_and_dirs("D:\src\github\ort-extensions\.scb\temp.win-amd64-cpython-311\Release\_deps\vcpkg\src\vcpkg\buildtrees\0.vcpkg_tags.cmake"): permission denied
+  # second attempt works, but that's not good enough for a CI.
+  add_dependencies(getopenssl getrapidjson)
+
+  add_dependencies(getcurl getopenssl)
+
+  set(triton_extra_cmake_args -DVCPKG_TARGET_TRIPLET=${vcpkg_triplet}
                               -DCMAKE_TOOLCHAIN_FILE=${VCPKG_SRC}/scripts/buildsystems/vcpkg.cmake)
-  set(triton_patch_command "")
-  set(triton_dependencies ${VCPKG_DEPENDENCIES})
 
+  # insane... but we added patch.exe from the git 32-bit windows distribution because the CI image has removed it
+  # from the git install that is available, and this seems like the least egregious/inconsistent way to make it work...
+  set(triton_patch_exe ${PROJECT_SOURCE_DIR}/cmake/externals/git.Win32.2.41.03.patch/patch.exe)
```

Comment:
should this patch tools in binary format be moved into some sub-folder in /tools?